metalearners/grid_search.py

```diff
@@ -312,7 +346,22 @@ def fit(
                     )
                 )
 
-        parallel = Parallel(n_jobs=self.n_jobs, verbose=self.verbose)
-        raw_results = parallel(delayed(_fit_and_score)(job) for job in jobs)
-        self.raw_results_ = raw_results
-        self.results_ = _format_results(results=raw_results)
+        self.grid_size_ = len(jobs)
+        self.raw_results_: list[GSResult] | Generator[GSResult, None, None] | None
+        self.results_: pd.DataFrame | None
```

Comment:
```suggestion
        self.results_: pd.DataFrame | None = None
```

Reply:
I didn't define it here because in ``sklearn``, the rule they follow is the fact that attributes that should be present after fitting (the ones finishing with a `_`) are not defined in the ``__init__`` rather just in `fit`, so in case the user calls them before fitting it raises an error instead of returning `None`. I checked the other classes and we don't fully follow this rule so I am open to changing it but it may be a thing to consider in the future.

Reply:
What about not using the trailing underscore to begin with?

Reply:
I think it may be interesting to leave the trailing underscore as these attributes should only be available after fitting and therefore should not be initialized before. I got confused and thought this line you were modifying was in the ``__init__`` and not in `fit`, I think we can set it to `None` as you suggested and remove the `else`.